auth/src/main/java/org/aerogear/mobile/security/impl/SecurityCheckExecutorImpl.java

```diff
@@ -0,0 +1,78 @@
+package org.aerogear.mobile.security.impl;
+
+import android.content.Context;
+
+import org.aerogear.mobile.core.metrics.MetricsService;
+import org.aerogear.mobile.security.SecurityCheckType;
+import org.aerogear.mobile.security.SecurityCheck;
+import org.aerogear.mobile.security.SecurityCheckExecutor;
+import org.aerogear.mobile.security.SecurityCheckResult;
+import org.aerogear.mobile.security.metrics.SecurityCheckResultMetric;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+
+public class SecurityCheckExecutorImpl implements SecurityCheckExecutor {
+    private final Collection<SecurityCheck> checks;
```

Comment:
why a `Collection`? instead of a `Set`? we can also initialise the thing on this line, instead of in the ctor

Reply:
I suggested to use a `Collection` as interface and an `HashSet` as implementation.
The reason is that we need to have a set of unique objects, but we don't need any of the methods exposed by a `Set` that are not exposed in a `Collection`. So, I think the most generic interface should be used, so that if we should ever decide to change from HashSet to ArrayList, for example, the rest of the code will stay unchanged.

Reply:
fair enough